Google.Cloud.AspNetCore/Google.Cloud.AspNetCore.DataProtection.Kms/GoogleCloudDataProtectionBuilderExtensions.cs

```diff
@@ -101,16 +101,15 @@ public static IDataProtectionBuilder ProtectKeysWithGoogleKms(this IDataProtecti
                         // but still use a GoogleCredential if it *has* been configured by a user, probably via reflection.
                         // This can be addressed when we've got the new auth library, which is a while off (as of 2019-04-03).
                         // (It's expected that by then we'll have a cleaner way of building a client with custom credentials, too.)
-                        GoogleCredential credential = services.GetService<GoogleCredential>();
+                        GoogleCredential credential = services.GetService<GoogleCredential>();                        
                         if (credential == null)
                         {
                             kmsClient = KeyManagementServiceClient.Create();
                         }
                         else
                         {
-                            var scopedCredential = credential.CreateScoped(KeyManagementServiceClient.DefaultScopes);
-                            var channel = new Channel(KeyManagementServiceClient.DefaultEndpoint.ToString(), scopedCredential.ToChannelCredentials());
-                            kmsClient = KeyManagementServiceClient.Create(channel);
+                            ICredential scopedCredential = credential.CreateScoped(KeyManagementServiceClient.DefaultScopes);
+                            kmsClient = new KeyManagementServiceClientBuilder { TokenAccessMethod = scopedCredential.GetAccessTokenForRequestAsync }.Build();
```

Comment:
I believe this should be:
```suggestion
                            kmsClient = new KeyManagementServiceClientBuilder { ChannelCredentials = scopedCredential.ToChannelCredentials() }.Build();
```

If you set the token directly it won't be refreshed when it has expired. If you set the credential, then the interceptor will be set on the channel, the interceptor is executed per call and each time it requests an access token from the credential, which the credential will refresh automatically if it has expired.

The TokenAccessMethod is for when you only have a token, but if you have a credential then you should always set that for automatic token refresh.

Reply:
This isn't setting a token - it's setting a method to call to *get* a token. (`GetAccessTokenForRequestAsync` is a method group.) The `ClientBuilderBase` creates an [`ITokenAccess` implementation](https://github.com/googleapis/gax-dotnet/blob/d1d6153fb52c96ca943facfd9d40f2e17eb0b6aa/Google.Api.Gax.Grpc/ClientBuilderBase.cs#L425) that calls the method any time it needs it, then calls `ToChannelCredentials` on that.

I'll change it as calling `ToChannelCredentials()` here is slightly less circuitous, but both should work.